pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -101,22 +101,24 @@ func Setup(mgr ctrl.Manager, args core.Args, l logging.Logger) error {
 		Complete(NewReconciler(mgr, dm,
 			WithLogger(l.WithValues("controller", name)),
 			WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
-			WithApplyOnceOnly(args.ApplyOnceOnly)))
+			WithApplyOnceOnly(args.ApplyOnceOnly),
+			WithApplyOnceOnlyForce(args.ApplyOnceOnlyForce)))
 }
 
 // An OAMApplicationReconciler reconciles OAM ApplicationConfigurations by rendering and
 // instantiating their Components and Traits.
 type OAMApplicationReconciler struct {
-	client        client.Client
-	components    ComponentRenderer
-	workloads     WorkloadApplicator
-	gc            GarbageCollector
-	scheme        *runtime.Scheme
-	log           logging.Logger
-	record        event.Recorder
-	preHooks      map[string]ControllerHooks
-	postHooks     map[string]ControllerHooks
-	applyOnceOnly bool
+	client             client.Client
+	components         ComponentRenderer
+	workloads          WorkloadApplicator
+	gc                 GarbageCollector
+	scheme             *runtime.Scheme
+	log                logging.Logger
+	record             event.Recorder
+	preHooks           map[string]ControllerHooks
+	postHooks          map[string]ControllerHooks
+	applyOnceOnly      bool
+	applyOnceOnlyForce bool
```

Comment:
```
type ApplyOnceOnly struct{
  Force bool
}
```

```suggestion
	applyOnceOnly      *ApplyOnceOnly
```

If the applyOnceOnly is nil, no apply-once-only enabled. If it's not nil, check if force flag is enabled.